### PR TITLE
[System.IO] In some cases '/' is not a valid root directory - fixes 60138

### PR DESCRIPTION
**Pull request by EgorBo**
https://bugzilla.xamarin.com/show_bug.cgi?id=60138
if `Environment.GetLogicalDrives` doesn't contain '/' item - `Path.GetPathRoot` should return the longest suitable drive `Environment.GetLogicalDrives` contains (currently it always returns '/').

for example `Environment.GetLogicalDrives`:

- "/foo"
- "/foo/boo"
- "/tmp"

then

- `GetPathRoot("/foo/some/dir")` => "/foo"
- `GetPathRoot("/foo/boo")` => "/foo/boo"
- `GetPathRoot("/foo/booo")` => "/foo"

not sure how to test such fixes - I need to mock `Environment.GetLogicalDrives`.

### CODE REVIEW

File: mcs/class/corlib/System.IO/Path.cs [hunk 1]
```diff
@@ -428,6 +428,9 @@ internal static string InsecureGetFullPath (string path)
 		internal static bool IsDirectorySeparator (char c) {
 			return c == DirectorySeparatorChar || c == AltDirectorySeparatorChar;
 		}
+		
+		private static string AppendDirectorySeparator (string path) =>
+			path.EndsWith (DirectorySeparatorCharAsString) ? path : path + DirectorySeparatorCharAsString;
```
Comment (marek-safar) [inline mcs/class/corlib/System.IO/Path.cs]: Should be stringcomparer.ordinal check or even better just char check

File: mcs/class/corlib/System.IO/Path.cs [hunk 2]
```diff
@@ -440,9 +443,25 @@ public static string GetPathRoot (string path)
 			if (!IsPathRooted (path))
 				return String.Empty;
 			
-			if (DirectorySeparatorChar == '/') {
-				// UNIX
-				return IsDirectorySeparator (path [0]) ? DirectorySeparatorStr : String.Empty;
+			if (DirectorySeparatorChar == '/') { // UNIX
+				// Most likely the root path will be '/' but there are cases (see https://bugzilla.xamarin.com/show_bug.cgi?id=60138)
+				// when there is no such drive in Environment.GetLogicalDrives () 
+				foreach (var drive in Environment.GetLogicalDrives ()) {
```
Comment (marek-safar) [inline mcs/class/corlib/System.IO/Path.cs]: Store `Environment.GetLogicalDrives ()` to local variable

**Comment by EgorBo** [comment]
@monojenkins build

**Comment by marek-safar** [comment]
build

**Comment by marek-safar** [comment]
Seems to break on all Linux lanes

**Comment by marek-safar** [comment]
@EgorBo what's status of this PR?

**Pull request closed without merging**